### PR TITLE
optimize text classification application

### PR DESCRIPTION
**Pull request by lugimzzz**
### PR types
Others 

### PR changes
Others 

### Description
优化文本分类application文档和代码，提高易用性

主要修改:
- 不再使用默认数据集，使用本地数据进行训练
- 默认模型改为ernie 3.0 medium（更新配置和相关评测数据）
- 将运行python脚本中经常需要更改配置参数写到命令行中
- 多标签和层次分类数据集更换(多标签：cail2019-婚姻家庭领域要素识别，层次分类：CMID医疗意图识别)

### CODE REVIEW

File: applications/text_classification/doccano.md [hunk 1]
```diff
@@ -190,7 +191,7 @@ python doccano.py \
 python doccano.py \
     --doccano_file doccano.jsonl \
     --save_dir ./data \
-    --splits 0.8 0.1 0.1 \
+    --splits 0.8 0.2 \
     --task_type "hierarchical" \
     --separator "##"
```
Comment (wawltor) [inline applications/text_classification/doccano.md]: 这里的分割符号是不是要改掉

File: applications/text_classification/hierarchical/README.md [hunk 2]
```diff
@@ -16,9 +18,9 @@
 
 ## 层次分类任务介绍
 
-多标签层次分类任务指自然语言处理任务中，每个样本具有多个标签标记，并且标签集合中标签之间存在预定义的层次结构，多标签层次分类需要充分考虑标签集之间的层次结构关系来预测层次化预测结果。层次分类任务中标签层次结构分为两类，一类为树状结构，另一类为有向无环图(DAG)结构。有向无环图结构与树状结构区别在于，有向无环图中的节点可能存在不止一个父节点。在现实场景中，大量的数据如新闻分类、专利分类、学术论文分类等标签集合存在层次化结构，需要利用算法为文本自动标注更细粒度和更准确的标签。
+多标签层次分类任务指自然语言处理任务中，**每个样本具有多个标签标记，并且标签集合中标签之间存在预定义的层次结构**，多标签层次分类需要充分考虑标签集之间的层次结构关系来预测层次化预测结果。层次分类任务中标签层次结构分为两类，一类为树状结构，另一类为有向无环图(DAG)结构。有向无环图结构与树状结构区别在于，有向无环图中的节点可能存在不止一个父节点。在现实场景中，大量的数据如新闻分类、专利分类、学术论文分类等标签集合存在层次化结构，需要利用算法为文本自动标注更细粒度和更准确的标签。
 
-层次分类问题可以被视为一个多标签问题，以下图一个树状标签结构(宠物为根节点)为例，如果一个样本属于美短虎斑，样本也天然地同时属于类别美国短毛猫和类别猫两个样本标签。本项目采用通用多标签层次分类算法，将每个结点的标签路径视为一个多分类标签，使用单个多标签分类器进行决策，以上面美短虎斑的例子为例，该样本包含三个标签：猫、猫##美国短毛猫、猫##美国短毛猫##美短虎斑(不同层的标签之间使用`##`作为分割符)。下图的标签结构标签集合为猫、猫##波斯猫、猫##缅因猫、猫##美国短毛猫、猫##美国短毛猫##美短加白、猫##美国短毛猫##美短虎斑、猫##美国短毛猫##美短起司、兔、兔##侏儒兔、兔##垂耳兔总共10个标签。
+层次分类问题是一个天然多标签问题，以下图一个树状标签结构(宠物为根节点)为例，如果一个样本属于美短虎斑，样本也天然地同时属于类别美国短毛猫和类别猫两个样本标签。本项目采用通用多标签层次分类算法，**将每个结点的标签路径视为一个多分类标签，使用单个多标签分类器进行决策**，以上面美短虎斑的例子为例，该样本包含三个标签：猫、猫##美国短毛猫、猫##美国短毛猫##美短虎斑(不同层的标签之间使用`##`作为分割符)。下图的标签结构标签集合为猫、猫##波斯猫、猫##缅因猫、猫##美国短毛猫、猫##美国短毛猫##美短加白、猫##美国短毛猫##美短虎斑、猫##美国短毛猫##美短起司、兔、兔##侏儒兔、兔##垂耳兔总共10个标签。
```
Comment (wawltor) [inline applications/text_classification/hierarchical/README.md]: 层次分类目前是逗号分割还是 ## 分割了

File: applications/text_classification/hierarchical/README.md [hunk 3]
```diff
@@ -57,133 +60,102 @@ hierarchical_classification/
...
+train.txt(训练数据集文件)， dev.txt(开发数据集文件)，test.txt(可选，测试数据集文件)，文件中文本与标签类别名，不同级的标签类别名用tab符`'\t'`分隔开，标签中多个标签之间用`','`逗号分隔开。
+
+训练集指用于训练模型的数据；开发集指用于评测模型表现的数据，可以根据模型在开发集上的精度调整训练参数和模型；测试集用于测试模型表现，没有测试集时可以使用开发集代替；通常建议训练集：开发集比例为8:2或7:3; 建议训练集、开发集、测试集的比例为8:1:1或6:2:2。**注意文本中不能包含tab符`'\t'`，对于第i层数据没有标签的，使用空字符`''`来表示**。
 
 - train.txt/dev.txt/test.txt 文件格式：
 ```text
-<输入序列1>'\t'<level 1 标签1>','<level 1 标签2>'\t'<level 2 标签1>','<level 2 标签2>'\t'...'\t'<level n 标签1>','<level n 标签2>
-<输入序列2>'\t'<level 1 标签>'\t'<level 2 标签>'\t'...'\t'<level n 标签>
+<文本1>'\t'<文本1：一级标签1>','<文本1：一级标签2>'\t'<文本2：二级标签1>','<文本1：二级标签2>','<文本1：二级标签3>
+<文本2>'\t'<文本2：一级标签1>'\t'<文本2：二级标签1>','<文本2：二级标签2>
```
Comment (wawltor) [inline applications/text_classification/hierarchical/README.md]: 这里比较好奇的是为什么层次分类中，不同层次上有不同的标签

File: applications/text_classification/doccano.md (hunk 1 above)
Comment (lugimzzz) [reply applications/text_classification/doccano.md]: 已去掉分割符号

File: applications/text_classification/hierarchical/README.md (hunk 2 above)
Comment (lugimzzz) [reply applications/text_classification/hierarchical/README.md]: 层次分类目前不同层次之间分割符还是 ##

File: applications/text_classification/hierarchical/README.md (hunk 3 above)
Comment (lugimzzz) [reply applications/text_classification/hierarchical/README.md]: 存在层次分类多标签路径的情况

File: applications/text_classification/hierarchical/README.md [hunk 4]
```diff
@@ -57,133 +60,103 @@ hierarchical_classification/
...
-
-程序运行时将会自动进行训练，评估，测试。同时训练过程中会自动保存开发集上最佳模型在指定的 `save_dir` 中，保存模型文件结构如下所示：
-
-```text
-checkpoint/
-├── model_config.json
-├── model_state.pdparams
-├── tokenizer_config.json
-└── vocab.txt
+pip install sklearn
```
Comment (wawltor) [inline applications/text_classification/hierarchical/README.md]: 看看 sklearn要不要限定一个版本

File: applications/text_classification/hierarchical/README.md [hunk 5]
```diff
@@ -192,54 +165,168 @@ data.txt(可选，待预测数据文件)。
...
+    --logging_steps 5
 ```
-使用多卡训练可以指定多个GPU卡号，例如 --gpus "0,1"
+使用多卡训练可以指定多个GPU卡号，例如 --gpus "0,1"。如果设备只有一个GPU卡号默认为0，可使用`nvidia-smi`命令查看GPU使用情况。
 
-更多数据集读取格式详见[数据集加载](https://paddlenlp.readthedocs.io/zh/latest/data_prepare/dataset_load.html#)和[自定义数据集](https://paddlenlp.readthedocs.io/zh/latest/data_prepare/dataset_self_defined.html)。
+可支持配置的参数：
 
-## 模型预测
+* `device`: 选用什么设备进行训练，选择cpu、gpu、xpu、npu。如使用gpu训练，可使用参数--gpus指定GPU卡号；默认为"gpu"。
```
Comment (wawltor) [inline applications/text_classification/hierarchical/README.md:217]: device应该是不能选定卡号的，因为 --gpus选定卡号的优先级的卡号更高

File: applications/text_classification/hierarchical/README.md [hunk 6]
```diff
@@ -278,63 +354,67 @@ pip install paddleslim==2.2.2
...
-  * `dataset`：训练数据集;默认为wos数据集。
-  * `dataset`：本地数据集路径，路径内需要包含train.txt, dev.txt, label.txt文件;默认为None。
-  * `depth`：层次分类数据标签最大深度;默认为2。
-  * `max_seq_length`：ERNIE/BERT模型使用的最大序列长度，最大不能超过512, 若出现显存不足，请适当调低这一参数；默认为512。
+  * `dataset_dir`：本地数据集路径，需包含train.txt,dev.txt,label.txt;默认为None。
+  * `max_seq_length`：模型使用的最大序列长度，建议与训练过程保持一致, 若出现显存不足，请适当调低这一参数；默认为128。
+
 * `ModelArguments`
   * `params_dir`：待预测模型参数文件；默认为"./checkpoint/"。
+  * `width_mult`：裁剪宽度保留的比例，表示对 `q`、`k`、`v` 以及 `ffn` 权重宽度的保留比例，默认是 '2/3'。
```
Comment (wawltor) [inline applications/text_classification/hierarchical/README.md]: 表示对 `q`、`k`、`v` 以及 `ffn`  -> self_attention中的 `q`、`k`、`v` 以及 `ffn` 权重宽度

File: applications/text_classification/hierarchical/deploy/predictor/infer.py [hunk 7]
```diff
@@ -28,15 +28,11 @@
                     required=True,
                     help="The path prefix of inference model to be used.")
 parser.add_argument("--model_name_or_path",
-                    default="ernie-2.0-base-en",
+                    default="ernie-3.0-medium-zh",
```
Comment (wawltor) [inline applications/text_classification/hierarchical/deploy/predictor/infer.py:31]: 这里看看要不要圈定一些模型的choice

File: applications/text_classification/multi_class/README.md [hunk 8]
```diff
@@ -51,109 +53,69 @@ multi_class/
...
+安装PaddleNLP默认开启百度镜像源来加速下载，如果您使用 HTTP 代理可以关闭(删去 -i https://mirror.baidu.com/pypi/simple)，更多关于PaddleNLP安装的详细教程请查见[PaddleNLP快速安装](https://github.com/PaddlePaddle/PaddleNLP/blob/develop/docs/get_started/installation.rst)。
 
 
-```shell
-wget https://paddlenlp.bj.bcebos.com/datasets/KUAKE_QIC.tar.gz
-tar -zxvf KUAKE_QIC.tar.gz
-mv KUAKE_QIC data
-```
+## 数据集准备
+训练需要准备指定格式的本地数据集,如果没有已标注的数据集，可以参考[文本分类任务doccano数据标注使用指南](https://github.com/PaddlePaddle/PaddleNLP/blob/develop/applications/text_classification/doccano.md)进行文本分类数据标注。
```
Comment (wawltor) [inline applications/text_classification/multi_class/README.md]: 这里改成相对位置的路径吧，其他的地方也修改一下

File: applications/text_classification/multi_class/README.md [hunk 9]
```diff
@@ -51,109 +53,69 @@ multi_class/
...
+**训练、开发、测试数据集**
 
-train.txt(训练数据集文件), dev.txt(开发数据集文件),输入文本序列与标签类别名用`'\t'`分隔开。
-- train.txt/dev.txt 文件格式：
+train.txt(训练数据集文件)， dev.txt(开发数据集文件)，test.txt(测试数据集文件)，文件中文本与标签类别名用tab符`'\t'`分隔开。训练集指用于训练模型的数据；开发集指用于评测模型表现的数据，可以根据模型在开发集上的精度调整训练参数和模型；测试集用于测试模型表现，没有测试集时可以使用开发集代替；通常建议训练集、开发集、测试集的比例为8:1:1或6:2:2；只有训练集和开发集的情况时建议训练集：开发集比例为8:2或7:3。**注意文本中不能包含tab符**
+- train.txt/dev.txt/test.txt 文件格式：
 ```text
-<输入序列1>'\t'<标签1>'\n'
-<输入序列2>'\t'<标签2>'\n'
+<文本1>'\t'<文本1:标签>
```
Comment (wawltor) [inline applications/text_classification/multi_class/README.md]: 这里有点怪怪的， 是不是 <文本1>'\t'<标签1>就行

File: applications/text_classification/multi_class/README.md [hunk 10]
```diff
@@ -51,109 +53,69 @@ multi_class/
...
+## 准备环境
+**文本分类所需用到的环境配置：**
 
-单卡训练
-```shell
-python train.py --warmup --early_stop --epochs 100
-```
+- python >= 3.6
+- paddlepaddle >= 2.2
+- paddlenlp >= 2.3
```
Comment (wawltor) [inline applications/text_classification/multi_class/README.md]: 建议paddle的版本升级2.3以上、paddlenlp的版本到2.3.4，建议层次分类、多标签分类都限定一下版本

File: applications/text_classification/multi_label/deploy/paddle_serving/README.md [hunk 11]
```diff
@@ -130,13 +134,15 @@ python rpc_client.py
...
+data:  被告于2016年3月将车牌号为皖B×××××出售了2.7万元，被告通过原告偿还了齐荷花人民币2.6万元，原、被告尚欠齐荷花2万元。
+label:  2,9
 --------------------
-data:  静乐县人民检察院指控，2014年8月30日15时许，静乐县苏坊村村民张某某因占地问题去苏坊村半切沟静静铁路第五标施工地点阻拦施工时，遭被告人王某某阻止，张某某打电话叫来儿子李某某，李某某看到张某某躺在地上，打了王某某一耳光。于是王某某指使工人殴打李某某，致李某某受伤。经忻州市公安司法鉴定中心鉴定，李某某的损伤评定为轻伤一级。李某某被打伤后，被告人王某某为逃避法律追究，找到任某某，指使任某某作实施××的伪证，并承诺每月给1万元。同时王某某指使工人王某甲、韩某某去丰润派出所作由任某某打伤李某某的伪证，导致任某某被静乐县公安局以涉嫌××罪刑事拘留。公诉机关认为，被告人王某某的行为触犯了《中华人民共和国刑法》××、《中华人民共和国刑法》××××之规定，应以××罪和××罪追究其刑事责任，数罪并罚。
-label:  0,61
+data:  2、判令被告返还借婚姻索取的现金33万元，婚前个人存款10万元；
+label:  10
 --------------------
+data:  一、判决原告于某某与被告杨某某离婚；
+label:  8,11
```
Comment (wawltor) [inline applications/text_classification/multi_label/deploy/paddle_serving/README.md:147]: 我看三个算法中，label 都是数字，这块能加载一下label，然后打印具体的label吗？

File: applications/text_classification/hierarchical/README.md (hunk 4 above)
Comment (lugimzzz) [reply applications/text_classification/hierarchical/README.md]: 限定- scikit-learn >= 0.24.2

File: applications/text_classification/hierarchical/README.md (hunk 6 above)
Comment (lugimzzz) [reply applications/text_classification/hierarchical/README.md]: 已修改

File: applications/text_classification/hierarchical/deploy/predictor/infer.py (hunk 7 above)
Comment (lugimzzz) [reply applications/text_classification/hierarchical/deploy/predictor/infer.py:31]: 中文用户可选ERNIE 3.0系列，英文用户可选ERNIE 2.0系列，默认为"ernie-3.0-medium-zh"
* `model_name`：选择预训练模型,可选"ernie-3.0-xbase-zh", "ernie-3.0-base-zh", "ernie-3.0-medium-zh", "ernie-3.0-micro-zh", "ernie-3.0-mini-zh", "ernie-3.0-nano-zh", "ernie-2.0-base-en", "ernie-2.0-large-en"；默认为"ernie-3.0-medium-zh"。

File: applications/text_classification/multi_class/README.md (hunk 8 above)
Comment (lugimzzz) [reply applications/text_classification/multi_class/README.md]: 均已修改

File: applications/text_classification/multi_class/README.md (hunk 9 above)
Comment (lugimzzz) [reply applications/text_classification/multi_class/README.md]: 已修改

File: applications/text_classification/multi_label/deploy/paddle_serving/README.md (hunk 11 above)
Comment (lugimzzz) [reply applications/text_classification/multi_label/deploy/paddle_serving/README.md:147]: 因为这个在服务部署中，暂不修改

File: applications/text_classification/multi_class/README.md (hunk 10 above)
Comment (lugimzzz) [reply applications/text_classification/multi_class/README.md]: 已修改
**文本分类所需用到的环境配置：**

- python >= 3.6
- paddlepaddle >= 2.3
- paddlenlp >= 2.3.4
- scikit-learn >= 0.24.2
